### PR TITLE
Adding UT for OS package

### PR DESCRIPTION
**Pull request by bardielle**

### CODE REVIEW

**Comment by jakub-dzon** [comment]
@bardielle - please fix the build failures.

**Review by jakub-dzon — CHANGES_REQUESTED** [review]

File: internal/os/os.go [hunk 1]
```diff
@@ -41,9 +38,10 @@ type OS struct {
 	LastUpgradeTime                 string
 	GracefulRebootChannel           chan struct{}
 	GracefulRebootCompletionChannel chan struct{}
+	OsExecCommands                  OsExecCommands
 }
 
-func NewOS(gracefulRebootChannel chan struct{}) *OS {
+func NewOS(gracefulRebootChannel chan struct{}, osExecCommands OsExecCommands) *OS {
```
Comment (masayag) [inline internal/os/os.go:44]: As mentioned earlier, we can skip the requirement for a second argument and initialize OsExecCommands in this method.

File: internal/os/os.go [hunk 2]
```diff
@@ -105,8 +104,8 @@ func (o *OS) Update(configuration models.DeviceConfigurationMessage) error {
 
 	if newOSInfo.HostedObjectsURL != o.HostedObjectsURL {
 		log.Infof("Hosted Images URL has been changed to %s", newOSInfo.HostedObjectsURL)
-		err := updateURLInEdgeConfFile(newOSInfo.HostedObjectsURL)
-		if err != nil {
+		err := o.OsExecCommands.UpdateUrlInEdgeRemote(newOSInfo.HostedObjectsURL, EdgeConfFileName)
+		if err != nil{
```
Comment (masayag) [inline internal/os/os.go:108]: nit: should be space between nil and {

File: internal/os/os_exec_commands.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+type OsExecCommands interface {
+	RpmOstreeStatus() ([]byte, error)
+	RpmOstreeUpdatePreview() ([]byte, error)
+	RpmOstreeUpgrade() error
+	SystemReboot() error
+	EnsureScriptExists(fileName string, script string) error
+	UpdateUrlInEdgeRemote(newURL string, remoteFileName string) error
+}
+
+type osExecCommandsStruct struct {}
```
Comment (masayag) [inline internal/os/os_exec_commands.go:24]: IINW the suffix `Struct` seems redundant. Can `osExecCommands` be used?

File: internal/os/os_test.go [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+	})
+
+	Context("Update OS status test", func() {
+
+		It("Get upgrade status happy flow", func() {
+			// given
+			deviceOS.RequestedOsCommit = RequestedCommitId
+
+			osExecCommandsMock.EXPECT().RpmOstreeStatus().Return(returnMockRpmOstreeStatus(RequestedCommitId, UpgradeTime, OldCommitID)).AnyTimes()
+
```
Comment (masayag) [inline internal/os/os_test.go:70]: can we set the number of calls to the expected amount instead of `AnyTimes`?

**Review by masayag — CHANGES_REQUESTED** [review]

File: internal/os/os.go [hunk 5]
```diff
@@ -41,9 +38,10 @@ type OS struct {
 	LastUpgradeTime                 string
 	GracefulRebootChannel           chan struct{}
 	GracefulRebootCompletionChannel chan struct{}
+	OsExecCommands                  OsExecCommands
```
[the remark this answers is not in the document]
Comment (bardielle) [reply internal/os/os.go:41]: But that's the way I can mock the functions in OsExecCommands

**Review by masayag — APPROVED** [review]

File: internal/os/os_test.go [hunk 6]
```diff
@@ -0,0 +1,340 @@
...
+			cfg := createConfig(true, OldCommitID, NewHostedUrl)
+			osExecCommandsMock.EXPECT().UpdateUrlInEdgeRemote(NewHostedUrl, os2.EdgeConfFileName).Return(nil)
+			osExecCommandsMock.EXPECT().RpmOstreeStatus().Return(returnMockRpmOstreeBootedOnlyStatus(OldCommitID, UpgradeTime))
+
+			// when
+			err := deviceOS.Update(cfg)
+
+			//then
+			Expect(err).NotTo(HaveOccurred())
+			upgradeStatusBeforeAfterUpdate := deviceOS.GetUpgradeStatus()
```
Comment (masayag) [inline internal/os/os_test.go:171]: s/upgradeStatusBeforeAfterUpdate/upgradeStatusAfterUpdate ?

File: internal/os/os_test.go [hunk 7]
```diff
@@ -0,0 +1,340 @@
...
+			osExecCommandsMock.EXPECT().RpmOstreeUpgrade().Return(nil)
+			osExecCommandsMock.EXPECT().SystemReboot().Return(nil)
+			waitForGracefulRebootChannel(deviceOS)
+
+			// when
+			err := deviceOS.Update(cfg)
+
+			//then
+			Expect(err).NotTo(HaveOccurred())
+			upgradeStatusBeforeAfterUpdate := deviceOS.GetUpgradeStatus()
```
Comment (masayag) [inline internal/os/os_test.go:193]: s/upgradeStatusBeforeAfterUpdate/upgradeStatusAfterUpdate ?

**Review by jakub-dzon — APPROVED** [review]

**Pull request merged**